src/lib/components/ui/tooltip.svelte

```diff
@@ -1,119 +1,157 @@
 <script lang="ts">
 	import type { Snippet } from 'svelte'
 	import type { HTMLAttributes } from 'svelte/elements'
-
 	type Position = 'top' | 'bottom' | 'left' | 'right'
 	interface Props extends HTMLAttributes<HTMLElement> {
 		helperText?: Snippet
 		position?: Position
 		large?: boolean
 		show?: boolean
+		tooltipId?: string
+		tooltipName: string
 	}
-	let { position = 'top', children, helperText, large = false, show = false }: Props = $props()
-	let tooltip: HTMLDivElement | undefined = $state(undefined)
-	let element: HTMLDivElement | undefined = $state(undefined)
-
-	$effect(() => {
-		// Update tooltip position when large or show changes
-		large
-		show
-
-		updateTooltipPosition()
-		window.addEventListener('resize', updateTooltipPosition)
-		window.addEventListener('scroll', updateTooltipPosition)
-
-		return () => {
-			window.removeEventListener('resize', updateTooltipPosition)
-			window.removeEventListener('scroll', updateTooltipPosition)
-		}
-	})
-	function updateTooltipPosition() {
-		if (tooltip && element) {
-			const elementRect = element.getBoundingClientRect()
-			const tooltipRect = tooltip.getBoundingClientRect()
-
-			const padding = 8
-			// Set tooltip initial position
-			let top = 0,
-				left = 0
-			if (position === 'top') {
-				top = elementRect.top - tooltipRect.height - padding
-				left = elementRect.left + elementRect.width / 2 - tooltipRect.width / 2
-			} else if (position === 'bottom') {
-				top += elementRect.bottom + padding
-				left = elementRect.left + elementRect.width / 2 - tooltipRect.width / 2
-			} else if (position === 'left') {
-				top = elementRect.top + elementRect.height / 2 - tooltipRect.height / 2
-				left = elementRect.left - tooltipRect.width - padding
-			} else if (position === 'right') {
-				top = elementRect.top + elementRect.height / 2 - tooltipRect.height / 2
-				left = elementRect.right + padding
-			}
-
-			// Check and adjust position if out of screen
-			if (top < 0) {
-				top = elementRect.bottom - padding
-			}
-			if (left < 0) {
-				left = elementRect.right - padding
-			}
-			if (top + tooltipRect.height > window.innerHeight) {
-				top = elementRect.top - tooltipRect.height + padding
-			}
-			if (left + tooltipRect.width > window.innerWidth) {
-				left = elementRect.left - tooltipRect.width + padding
-			}
-
-			tooltip.style.top = `${top}px`
-			tooltip.style.left = `${left}px`
-		}
-	}
+	let {
+		position = 'top',
+		children,
+		helperText,
+		large = false,
+		show = false,
+		tooltipId = Math.random().toString(16),
+		tooltipName = 'tooltip',
```

Comment:
Can this be randomised too?
```suggestion
		tooltipName = 'tooltip' + Math.random().toString(16),
```